trin-execution/src/subcommands/era2.rs

```diff
@@ -0,0 +1,243 @@
+                    storage_index_hash,
+                    value,
+                } in storage_entry.0
+                {
+                    account_trie
+                        .insert(storage_index_hash.as_slice(), &alloy_rlp::encode(value))?;
+                }
+            }
+
+            if account_trie.root_hash()? != account_state.storage_root {
+                return Err(Error::msg("Failed importing account storage trie: storage roots don't match expect value, .era2 import failed"));
+            }
+
+            // Insert contract if available
+            if !bytecode.is_empty() && account_state.code_hash != KECCAK_EMPTY {
+                ensure!(
+                    account_state.code_hash == keccak256(&bytecode),
+                    "Code hash mismatch, .era2 import failed"
+                );
+                self.trin_execution
+                    .database
+                    .db
+                    .put(keccak256(&bytecode), bytecode.clone())?;
+            }
+
+            // Insert account into state trie
+            self.trin_execution
+                .database
+                .trie
+                .lock()
+                .insert(address_hash.as_slice(), &alloy_rlp::encode(&account_state))?;
+
+            self.trin_execution
+                .database
+                .db
+                .put(address_hash, alloy_rlp::encode(account_state))
+                .expect("Inserting account should never fail");
+
+            accounts_imported += 1;
+            if accounts_imported % 1000 == 0 {
+                info!("Imported {} accounts", accounts_imported);
+                info!("Committing changes to database");
+                self.trin_execution.get_root()?;
+                info!("Finished committing changes to database");
+            }
+        }
+
+        // Check if the state root matches, if this fails it means either the .era2 is wrong or we
+        // imported the state wrong
+        if era2.header.header.state_root != self.trin_execution.get_root()? {
+            return Err(Error::msg("State root mismatch, .era2 import failed"));
+        }
+
+        // Save execution position
+        self.trin_execution
+            .execution_position
+            .update_position(self.trin_execution.database.db.clone(), era2.header.header)?;
+
+        info!("Done importing State from .era2 file");
+
+        Ok(())
+    }
+
+    /// insert the last 256 block hashes into the database
+    pub async fn import_last_256_block_hashes(&mut self) -> anyhow::Result<()> {
+        let first_block_hash_to_add = self
+            .trin_execution
+            .next_block_number()
+            .saturating_sub(BLOCKHASH_SERVE_WINDOW);
+        let mut era_manager = EraManager::new(first_block_hash_to_add).await?;
+        for block_number in first_block_hash_to_add..self.trin_execution.next_block_number() {
+            let block = era_manager.get_next_block().await?;
+            ensure!(
```

Comment:
nit: I'm not sure how I feel about this check. On one side, it's good to double check, on the other side, we rely on `EraManager` returning blocks in consecutive order in other places.

If it was me, I would have trusted that `EraManager` does the correct job, and just do something like this:

```rust
while era_manager.next_block_number() < self.trin_execution.next_block_number() {
    let block = era_manager.get_next_block().await?;
    self.trin_execution.database.db.put(...);
}
```

Up to you.